java/code/src/com/redhat/rhn/common/db/datasource/xml/Package_queries.xml

```diff
@@ -1135,46 +1149,49 @@ SELECT 1
 
 <mode name="all_packages_in_channel_after" class="com.redhat.rhn.frontend.dto.PackageDto">
   <query params="cid, start_date_str">
-SELECT
-       P.id AS id,
-       PN.name as name,
-       NVL(PE.version, ' ') as version,
-       NVL(PE.release, ' ') as release,
-       NVL(PE.epoch, ' ') as epoch,
-       PA.label AS arch_label,
-       TO_CHAR(P.last_modified, 'YYYY-MM-DD HH24:MI:SS') AS last_modified
-  FROM rhnPackageArch PA, rhnPackageName PN, rhnPackageEVR PE, rhnPackage P,
-       rhnChannelPackage CP
- WHERE CP.channel_id = :cid
-   AND CP.package_id = P.id
-   AND P.name_id = PN.id
-   AND P.evr_id = PE.id
-   AND PA.id = P.package_arch_id
-   AND P.last_modified &gt;= TO_TIMESTAMP(:start_date_str, 'YYYY-MM-DD HH24:MI:SS')
+SELECT P.id AS id,
+      PN.name as name,
+      NVL(PE.version, ' ') as version,
+      NVL(PE.release, ' ') as release,
+      NVL(PE.epoch, ' ') as epoch,
+      PA.label AS arch_label,
+      TO_CHAR(P.last_modified, 'YYYY-MM-DD HH24:MI:SS') AS last_modified,
+      CV.checksum as checksum,
+      CV.checksum_type as checksum_type,
+      (CP.is_retracted = 1) as retracted
```

Comment:
`(CP.is_retracted = 1)` should this turn the result to "boolean"?
I am not sure but I think 1/0 should do the same job.
Is this really needed?

Reply:
Are you sure? I thought `1/0` will appear as an integer in the result. Here we need a boolean since the `PackageDto` bean has a boolean attribute `retracted`. The best way would be to re-write all queries in this PR to return a boolean, not `1` or `0`. Let me do it, because I also promised that @lucidd .

Reply:
We come from orcale where "boolean" did not exist. So I think the conversion just works.

See SsmUpgradablePackageListItem.java:
```
    private boolean errataRestart;
    public void setErrataRestart(boolean errataRestartIn) {
        this.errataRestart = errataRestartIn;
    }
```

java/code/src/com/redhat/rhn/common/db/datasource/xml/Package_queries.xml:
```
<mode name="ssm_packages_for_upgrade" class="com.redhat.rhn.frontend.dto.SsmUpgradablePackageListItem">
...
        CASE WHEN RB.keyword IS NOT NULL THEN 1 ELSE 0 END AS errata_reboot,
        CASE WHEN RS.keyword IS NOT NULL THEN 1 ELSE 0 END AS errata_restart,
...
```

Reply:
I re-wrote all the queries so that they return boolean. I think this makes the most sense, because we are interested in a boolean information.